HyperVExtension/src/HyperVExtension/Helpers/DevSetupAgentDeploymentHelper.cs

```diff
@@ -0,0 +1,306 @@
+        }
+
+        return (ushort)psObject.BaseObject;
+    }
+
+    private readonly string _script = @"
+function DeployDevSetupAgent
+{
+    Param(
+        [Guid] $VMId,
+        [System.Management.Automation.Runspaces.PSSession] $Session,
+        [string] $Path
+    )
+
+        $ErrorActionPreference = ""Stop""
+        $activity = ""Installing DevSetupAgent to VM $VMId""
+
+        # Validate input. Only .cab and .zip files are supported
+        # If $Path is a directory, it will be copied to the VM and installed as is
+        $isDirectory = $false
+        $isCab = $false
+        $inputFileName = $null
+        if (Test-Path -Path $Path -PathType 'Container')
+        {
+            $isDirectory = $true
+        }
+        elseif (Test-Path -Path $Path -PathType 'Leaf')
+        {
+            if ($Path -match '\.(cab)$')
+            {
+                $isCab = $true
+            }
+            elseif (-not $Path -match '\.(zip)$')
+            {
+                throw ""Only .cab and .zip files are supported""
+            }
+            $inputFileName = Split-Path -Path $Path -Leaf
+        }
+        else
+        {
+            throw ""$Path does not exist""
+        }
+
+
+        $DevSetupAgentConst = ""DevSetupAgent""
+        $DevSetupEngineConst = ""DevSetupEngine""
+        $session = $Session
+
+        $guestTempDirectory = Invoke-Command -Session $session -ScriptBlock { $env:temp }
+        
+        [string] $guid = [System.Guid]::NewGuid()
+        $guestUnpackDirectory = Join-Path -Path $guestTempDirectory -ChildPath $guid
+        $guestDevSetupAgentTempDirectory = Join-Path -Path $guestUnpackDirectory -ChildPath $DevSetupAgentConst
+
+        Write-Host ""Creating VM temporary folder $guestUnpackDirectory""
+        Write-Progress -Activity $activity -Status ""Creating VM temporary folder $guestUnpackDirectory"" -PercentComplete 10
+        Invoke-Command -Session $session -ScriptBlock { New-Item -Path ""$using:guestUnpackDirectory"" -ItemType ""directory"" }
+
+        if ($isDirectory)
+        {
+            $destinationPath = $guestDevSetupAgentTempDirectory
+        }
+        else
+        {
+            $destinationPath = $guestUnpackDirectory
+        }
+
+        Write-Host ""Copying $Path to VM $destinationPath""
+        Write-Progress -Activity $activity -Status ""Copying DevSetupAgent to VM $destinationPath"" -PercentComplete 15
+        Copy-Item -ToSession $session -Recurse -Path $Path -Destination $destinationPath
+
+
+        Invoke-Command -Session $session -ScriptBlock {
+            $ErrorActionPreference = ""Stop""
+
+            try
+            {
+                $guestDevSetupAgentPath = Join-Path -Path $Env:Programfiles -ChildPath $using:DevSetupAgentConst
+
+                # Stop and remove previous version of DevSetupAgent service if it exists
+                $service = Get-Service -Name $using:DevSetupAgentConst -ErrorAction SilentlyContinue
+                if ($service -ne $null)

```

Comment:
Its best practice to put the $null on the left side of the comparison for PowerShell:
https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-null?view=powershell-7.2#checking-for-null , so I think we can update it here and also anywhere below.

```Suggestion
                if ($null -ne $service)
```

Reply:
Seems like it should be just "if ($service)".